### ISSUE TITLE
Per-Syntax settings

### ISSUE THREAD

**Issue by spywhere** [Microsoft/vscode#3533]
**TL;DR:** Add settings support for per-syntax or improve the Markdown syntax settings recognition.

**Problems:**
In my project development, I usually works across different file types. Each of them is work a bit different than the another.

Let say I have a `.js` (JavaScript) file and some `.md` (Markdown) file. When I works with `.js` file, I want to trim of the whitespace at the end of each line while when I works with `.md` file, I want the VSCode to keep the whitespace at the end of line as-is () since the markdown syntax use double-spaces at the end of line to indicate line break. When I enabled `files.trimTrailingWhitespace`, I cannot save the file without remove these trailing whitespace on `.md` file and that is annoying to me.

Also, this is just a simple usage example of this feature. There are many cases that this feature would solve them easily.

**Possible Solutions:**
- Implement per-syntax settings so each file type can be configured differently (preferred).
- Implement the "smart settings" where if you works with (let says) Markdown file and there are double-spaces at the end of line (or just any language feature at similar to this) with `files.trimTrailingWhitespace` turn on, VSCode will not trim those lines.

**Comment by joaomoreno** [comment]
cc @chrisdias @egamma @waderyan @seanmcbreen @bpasero @alexandrudima

**Comment by egamma** [comment]
@spywhere can you pls try [vscode-editorconfig extension](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) and see whether it covers your needs.

**Comment by spywhere** [comment]
@egamma That's cover my need. However, I didn't see it working on my project. I hope I'm not configured it wrong.

**Comment by bpasero** [comment]
https://github.com/Microsoft/vscode/issues/1587
